File: 918-maximum-sum-circular-subarray/918-maximum-sum-circular-subarray.cpp
class Solution {
public:
    
    int maxSubarray(vector<int>& nums){
        int mx_flg = 0;
        int mx_sum = INT_MIN;
        for(int i=0; i<nums.size(); i++){
            mx_flg = max(mx_flg+nums[i], nums[i]);
            mx_sum = max(mx_sum, mx_flg);
        }
        return mx_sum;
    }
    
    int maxSubarraySumCircular(vector<int>& nums) {
        int sum_arr = accumulate(nums.begin(), nums.end(), 0);
        int max_sum = maxSubarray(nums);
        transform(nums.begin(), nums.end(), nums.begin(), bind1st(std::multiplies<int>(), -1));
        int min_sum = maxSubarray(nums)*-1;
        cout<<"mx sum"<<max_sum<<"\n";
        cout<<"min sum"<<min_sum;
        return max_sum>0? max(max_sum, (sum_arr - min_sum)): max_sum;
    }
};

File: 918-maximum-sum-circular-subarray/918-maximum-sum-circular-subarray.java
class Solution {
    
    public int maxSubarraySum(int[] nums){
        int mxFlag = 0;
        int mxSum = Integer.MIN_VALUE;
        for(int i=0; i<nums.length; i++){
            mxFlag = Math.max(nums[i]+mxFlag, nums[i]);
            mxSum = Math.max(mxSum, mxFlag);
        }
        return mxSum;
    }
    
    public int maxSubarraySumCircular(int[] nums) {
        int sumSubarr = IntStream.of(nums).sum();
        int maxSum = maxSubarraySum(nums);
        int minSum = maxSubarraySum(Arrays.stream(nums).map(i -> i * -1).toArray())*-1;
        return maxSum>0? Math.max(maxSum, Math.max(maxSum, sumSubarr-minSum)): maxSum;
    }
}